File: ruby/1-BinaryGap.rb
# https://codility.com/programmers/lessons/1-iterations/binary_gap/start/

def solution(n)
  base = n.to_s(2).gsub(/0+$/, "")
  candidates = base.split("1").select{|i| i!= "" }
  
  return 0 if candidates.length < 1
  
  candidates.map{ |candidate|
    candidate.length
  }.max  
end

p solution(529)
p solution(15)
p solution(2147483647)

File: js/1-BinaryGap.js
// https://codility.com/programmers/lessons/1-iterations/binary_gap/start/

const solution = (N) => {
  let twoDig = parseInt(N, 10).toString(2);
  let result = 0;

  twoDig = twoDig.replace(/0+$/, '');

  const candidates = twoDig.split('1').filter((n) => {
    if (result < n.length) {
      result = n.length;
    }

    return n;
  });

  if (candidates.length < 1) {
    return 0;
  }

  return result;
};

const s = solution(15);
console.log(s);
